cmd/gcp-controller-manager/node_syncer.go

```diff
@@ -131,23 +139,50 @@ func (ns *nodeSyncer) processNext() bool {
 	return true
 }
 
+func (ns *nodeSyncer) workPodRemoveQueue() {
+	for ns.processNextPodRemove() {
+	}
+}
+
+func (ns *nodeSyncer) processNextPodRemove() bool {
+	key, quit := ns.podRemoveQueue.Get()
+	if quit {
+		return false
+	}
+	defer ns.podRemoveQueue.Done(key)
+
+	uid, ok := key.(types.UID)
+	if !ok {
+		klog.Warningf("Dropping invalid key %q for delayed pod removal", key)
+		return true
+	}
+	if err := ns.removePod(uid); err != nil {
```

Comment:
We don't re-queue here, but just log the error. So the error is not an actual error. Maybe we can let ns.removePod() return nothing and do the logging inside removePod().

Reply:
As discussed, I prefer not dumping log messages from removePod() which I consider as a utility method that can be used for other purposes.